internal/elasticsearch/ingest/processors_test.go

```diff
@@ -24,61 +24,104 @@ func TestResource_Processors(t *testing.T) {
 			content: []byte(`---
 description: Made up pipeline
 processors:
-# First processor.
-- grok:
-    tag: Extract header
-    field: message
-    patterns:
-    - \[%{APACHE_TIME:apache.error.timestamp}\] \[%{LOGLEVEL:log.level}\]( \[client
-      %{IPORHOST:source.address}(:%{POSINT:source.port})?\])? %{GREEDYDATA:message}
-    - \[%{APACHE_TIME:apache.error.timestamp}\] \[%{DATA:apache.error.module}:%{LOGLEVEL:log.level}\]
-      \[pid %{NUMBER:process.pid:long}(:tid %{NUMBER:process.thread.id:long})?\](
-      \[client %{IPORHOST:source.address}(:%{POSINT:source.port})?\])? %{GREEDYDATA:message}
-    pattern_definitions:
-      APACHE_TIME: '%{DAY} %{MONTH} %{MONTHDAY} %{TIME} %{YEAR}'
-    ignore_missing: true
+    - grok:
+        tag: Extract header
+        field: message
+        patterns:
+        - \[%{APACHE_TIME:apache.error.timestamp}\] \[%{LOGLEVEL:log.level}\]( \[client%{IPORHOST:source.address}(:%{POSINT:source.port})?\])? %{GREEDYDATA:message}
+        - \[%{APACHE_TIME:apache.error.timestamp}\] \[%{DATA:apache.error.module}:%{LOGLEVEL:log.level}\]
+        ignore_missing: true
 
-- date:
-    field: apache.error.timestamp
-    target_field: '@timestamp'
-    formats:
-    - EEE MMM dd H:m:s yyyy
-    - EEE MMM dd H:m:s.SSSSSS yyyy
-    on_failure:
-    - append:
+    - date:
+        field: apache.error.timestamp
+        target_field: '@timestamp'
+        formats:
+        - EEE MMM dd H:m:s yyyy
+        - EEE MMM dd H:m:s.SSSSSS yyyy
+        on_failure:
+        - append:
+            field: error.message
+            value: '{{ _ingest.on_failure_message }}'
+    - set:
+        description: Set event category
+        field: event.category
+        value: web
+    # Some script
+    - script:
+        lang: painless
+
+    - grok:
+        field: source.address
+        ignore_missing: true
+        patterns:
+        - ^(%{IP:source.ip}|%{HOSTNAME:source.domain})$
+    - rename:
+        field: source.as.organization_name
+        target_field: source.as.organization.name
+        ignore_missing: true
+on_failure:
+    - set:
         field: error.message
         value: '{{ _ingest.on_failure_message }}'
-- set:
-    description: Set event category
-    field: event.category
-    value: web
-# Some script
-- script:
-    lang: painless
-    source: >-
-      [...]
+`),
+			expected: []Processor{
+				{Type: "grok", FirstLine: 4, LastLine: 11},
+				{Type: "date", FirstLine: 12, LastLine: 21},
+				{Type: "set", FirstLine: 22, LastLine: 26},
```

Comment:
Line 26 is a comment, I think this should be also left unmarked.
```suggestion
				{Type: "set", FirstLine: 22, LastLine: 25},
```

Reply:
Ok, looks like a good option, I don't have a simpler alternative, it looks like the library doesn't help finding the last parsed line for each node.